_pytest/config.py

```diff
@@ -990,11 +990,15 @@ def pytest_load_initial_conftests(self, early_config):
 
     def _initini(self, args):
         ns, unknown_args = self._parser.parse_known_and_unknown_args(args, namespace=self.option.copy())
-        r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn)
+        rootdir = ns.rootdir if ns.rootdir else None
+        r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn, rootdir_cmd_arg=rootdir)
         self.rootdir, self.inifile, self.inicfg = r
         self._parser.extra_info['rootdir'] = self.rootdir
         self._parser.extra_info['inifile'] = self.inifile
         self.invocation_dir = py.path.local()
+        if ns.rootdir:
+            self.invocation_dir = self.rootdir
+            sys.path.append(str(self.rootdir))
```

Comment:
This seems strange to me, why is this change required?

Reply:
> what is the problem that you are encountering exactly

I want to write negative test and I can't match result. It's can't catch

> This seems strange to me, why is this change required?
```
            for path in sys.path:
                print(path)
```
```
/home/feuillemorte/github/pytest_fork/env/bin
/home/feuillemorte/github/pytest_fork/env/lib/python36.zip
/home/feuillemorte/github/pytest_fork/env/lib/python3.6
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/lib-dynload
/usr/lib64/python3.6
/usr/lib/python3.6
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages/pluggy-0.6.0-py3.6.egg
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages/attrs-17.4.0-py3.6.egg
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages/six-1.11.0-py3.6.egg
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages/py-1.5.2-py3.6.egg
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages/pytest-3.3.3.dev50+g83034bbd.d20180122-py3.6.egg
/home/feuillemorte/github/pytest_fork/env/lib/python3.6/site-packages
```

Python paths are not contains rootdir, so, I get an error:
`ModuleNotFoundError: No module named 'spoon'`

if it's not a good solution, please, give me an advice
Thank you!

Reply:
> and I can't to write test on it because

If that's because `result.stdout.str()` is returning an empty string it might be that you have to look for the error in `result.stderr.str()` instead. 

Perhaps you can post the exact test you're trying and what you expected? That way I might be able to point the exact problem. 

> Python paths are not contain rootdir

Oh you mean in the test? If that's the case you can change the *test* instead to include the path by calling `testdir.syspathinsert()` at the start of the test.

Reply:
> Oh you mean in the test

It's not only in test. It repeate when I test locally a rootdir option

Reply:
But the `rootdir` is not added currently, so I don't think it should be added if `--rootdir` is passed in the command-line either.

Reply:
I can check rootdir path and add to sys.path if it's not exist. Is it ok?

Reply:
I mean if you have tests in another folder, for example, `/home/user/some_dir/tests` you don't have this folder in paths. So, you can't import some files by this path if you pass arg `--rootdir=/home/user/some_dir/tests` - you'll get an import error

Reply:
@feuillemorte I see what you mean, but "automatically adding rootdir to sys.path" is a different issue than "set the rootdir on the command-line"; I don't think we should do both changes in the same PR and without some serious consideration.

Adding the `rootdir` to `sys.path` is not something to be done lightly because this can cause a lot of unexpected consequences, believe me. 😬

Reply:
I see, thank you. But really, I don't know in which cases rootdir will be used because we can't use imports there (except system imports
I'll delete this line then :)

Reply:
Only if import will be like 'from root.spoon...'
And in this case we can run like 'pytest root/tests' without rootdir option. Am I wrong?

Reply:
The `rootdir` is used so pytest can have a reproducible node ids between runs in different computers. Here's an excerpt [from the docs](https://docs.pytest.org/en/latest/customize.html?highlight=rootdir#initialization-determining-rootdir-and-inifile):

> * Construct nodeids during collection; each test is assigned a unique nodeid which is rooted at the rootdir and takes in account full path, class name, function name and parametrization (if any).
> * Is used by plugins as a stable location to store project/test run specific information; for example, the internal cache plugin creates a .cache subdirectory in rootdir to store its cross-test run state.

So it is not really related to `sys.path` at all... but I understand why it might be confusing given its name.

Reply:
I'll fix it asap )) I think today/tomorrow evening